android/app/src/androidTest/java/org/fossasia/openevent/DateTest.java

```diff
@@ -0,0 +1,59 @@
+package org.fossasia.openevent;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * Created by arpitdec5 on 21-05-2017.
+ */
+
+public class DateTest {
+
+    @Test
+    public void checkDate(){
+
+        //getting time in gmt
+        String gmt_time = "2014-04-02T07:59:02.111Z";
+        SimpleDateFormat readDate = new SimpleDateFormat("hh:mm a", Locale.getDefault());
+        readDate.setTimeZone(TimeZone.getTimeZone("GMT"));
+        try {
+
+            //checking time in gmt
+            Date gmt_date_1 = readDate.parse(gmt_time);
+            Assert.assertEquals("07:59:02", gmt_date_1);
+
+            //checking time in ist
+            readDate.setTimeZone(TimeZone.getTimeZone("IST"));
+            Date ist_date = readDate.parse(gmt_time);
+            Assert.assertEquals("13:29:02", ist_date);
+
+            //checking time in gmt + 4
+            readDate.setTimeZone(TimeZone.getTimeZone("GMT+04:00"));
+            Date gmt_date_2 = readDate.parse(gmt_time);
+            Assert.assertEquals("11:59:02", gmt_date_2);
+
+            //checking base case 12:00:00
+            readDate.setTimeZone(TimeZone.getTimeZone("GMT+04:01"));
+            Date gmt_date_3 = readDate.parse(gmt_time);
+            Assert.assertEquals("12:00:02", gmt_date_3);
+
+            //checking base case 00:00:00
+            readDate.setTimeZone(TimeZone.getTimeZone("GMT+16:01"));
+            Date gmt_Date_4 = readDate.parse(gmt_time);
+            Assert.assertEquals("00:00:02", gmt_Date_4);
+
+        }catch (ParseException e){
```

Comment:
OK. So what you can do is increase a parameter in each method, taking in a timezone and create helper method for others that by default call getTimezone()

This generally happens while testing

Second option is using Mockito, which returns what you want when getTimezone() is called, but it will be a little difficult, so use first method